From: Juan Navarro <juan.navarro@gmx.es>
Date: Wed, 21 Oct 2020 13:36:06 +0200
Subject: Revert "agent: add nice_agent_get_sockets API"

This reverts commit 875a23a72257f38a3286315e9752fe93e5ab9591.

The reason for reverting this commit is that it bumps the GLib
version requirement from 2.48 (as provided by Ubuntu 16.04) to
2.54 (Ubuntu 18.04). This means that it is not possible any more
to build libnice on Ubuntu 16.04.

This commit also introduces some new API calls that are NOT used
by Kurento, so we won't really notice not having them:
nice_agent_get_sockets
nice_component_get_sockets
---
 README                                      |  2 +-
 agent/agent.c                               | 15 ---------------
 agent/agent.h                               | 23 -----------------------
 agent/component.c                           | 27 ++++-----------------------
 agent/component.h                           |  2 --
 docs/reference/libnice/libnice-sections.txt |  1 -
 meson.build                                 |  2 +-
 nice/libnice.sym                            |  1 -
 8 files changed, 6 insertions(+), 67 deletions(-)

diff --git a/README b/README
index b307b9b..6180a0c 100644
--- a/README
+++ b/README
@@ -16,7 +16,7 @@ See the file COPYING.
 Requirements
 ------------
 
- glib >= 2.54
+ glib >= 2.44
  pkg-config
  gnutls >= 2.12.0 or OpenSSL
  gupnp-igd >= 0.1.2 (optional)
diff --git a/agent/agent.c b/agent/agent.c
index e642b63..c741046 100644
--- a/agent/agent.c
+++ b/agent/agent.c
@@ -6988,18 +6988,3 @@ nice_agent_close_async (NiceAgent *agent, GAsyncReadyCallback callback,
 
   agent_unlock (agent);
 }
-
-
-NICEAPI_EXPORT GPtrArray *
-nice_agent_get_sockets (NiceAgent *agent, guint stream_id, guint component_id)
-{
-  GPtrArray *array = NULL;
-  NiceComponent *component;
-
-  agent_lock (agent);
-  if (agent_find_component (agent, stream_id, component_id, NULL, &component))
-    array = nice_component_get_sockets (component);
-  agent_unlock (agent);
-
-  return array;
-}
diff --git a/agent/agent.h b/agent/agent.h
index 1164138..c3b77a3 100644
--- a/agent/agent.h
+++ b/agent/agent.h
@@ -1681,29 +1681,6 @@ void
 nice_agent_close_async (NiceAgent *agent, GAsyncReadyCallback callback,
     gpointer callback_data);
 
-/**
- * nice_agent_get_sockets:
- * @agent: The #NiceAgent Object
- * @stream_id: The ID of the stream
- * @component_id: The ID of the component
- *
- * Each component can have multiple sockets, this is an API to retrieve them all
- * to be able to set properties. Most of the sockets for a component are created when
- * calling nice_agent_gather_candidates(), so this API should be called right after to
- * able to set properties on the sockets before they are used.
- *
- * These sockets can be a mix of UDP & TCP sockets depending on the compatibility mode
- * and options that have been set.
- *
- * Returns: (element-type GSocket) (transfer full): An array
- * containing all of the sockets for this component. Free with
- * g_ptr_array_unref() when done.
- *
- * Since: 0.1.17
- */
-GPtrArray *
-nice_agent_get_sockets (NiceAgent *agent, guint stream_id, guint component_id);
-
 G_END_DECLS
 
 #endif /* __LIBNICE_AGENT_H__ */
diff --git a/agent/component.c b/agent/component.c
index b0e9fe0..f0bd1dc 100644
--- a/agent/component.c
+++ b/agent/component.c
@@ -451,7 +451,7 @@ nice_component_restart (NiceComponent *cmp)
 	nice_candidate_free (cmp->restart_candidate);
       cmp->restart_candidate = candidate;
     }
-    else 
+    else
       nice_candidate_free (candidate);
   }
   g_slist_free (cmp->remote_candidates),
@@ -469,7 +469,7 @@ nice_component_restart (NiceComponent *cmp)
 /*
  * Changes the selected pair for the component to 'pair'. Does not
  * emit the "selected-pair-changed" signal.
- */ 
+ */
 void
 nice_component_update_selected_pair (NiceAgent *agent, NiceComponent *component, const CandidatePair *pair)
 {
@@ -510,7 +510,7 @@ nice_component_update_selected_pair (NiceAgent *agent, NiceComponent *component,
 }
 
 /*
- * Finds a remote candidate with matching address and 
+ * Finds a remote candidate with matching address and
  * transport.
  *
  * @return pointer to candidate or NULL if not found
@@ -528,7 +528,7 @@ nice_component_find_remote_candidate (NiceComponent *component, const NiceAddres
       return candidate;
 
   }
-  
+
   return NULL;
 }
 
@@ -1622,22 +1622,3 @@ nice_component_verify_remote_candidate (NiceComponent *component,
 
   return FALSE;
 }
-
-/* Must be called with agent lock held */
-/* Returns a transfer full GPtrArray of GSocket */
-GPtrArray *
-nice_component_get_sockets (NiceComponent *component)
-{
-  GPtrArray *array = g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
-  GSList *item;
-
-  for (item = component->local_candidates; item; item = item->next) {
-    NiceCandidateImpl *c = item->data;
-    NiceSocket *nicesock = c->sockptr;
-
-    if (nicesock->fileno && !g_ptr_array_find (array, nicesock->fileno, NULL))
-      g_ptr_array_add (array, g_object_ref (nicesock->fileno));
-  }
-
-  return array;
-}
diff --git a/agent/component.h b/agent/component.h
index a36cc94..deae9b7 100644
--- a/agent/component.h
+++ b/agent/component.h
@@ -310,8 +310,6 @@ gboolean
 nice_component_verify_remote_candidate (NiceComponent *component,
     const NiceAddress *address, NiceSocket *nicesock);
 
-GPtrArray *
-nice_component_get_sockets (NiceComponent *component);
 
 G_END_DECLS
 
diff --git a/docs/reference/libnice/libnice-sections.txt b/docs/reference/libnice/libnice-sections.txt
index f5c13d7..7a4485a 100644
--- a/docs/reference/libnice/libnice-sections.txt
+++ b/docs/reference/libnice/libnice-sections.txt
@@ -54,7 +54,6 @@ nice_agent_parse_remote_stream_sdp
 nice_agent_parse_remote_candidate_sdp
 nice_agent_get_io_stream
 nice_agent_get_selected_socket
-nice_agent_get_sockets
 nice_agent_get_component_state
 nice_agent_close_async
 nice_component_state_to_string
diff --git a/meson.build b/meson.build
index 5e89b5e..a2c2728 100644
--- a/meson.build
+++ b/meson.build
@@ -22,7 +22,7 @@ endif
 soversion = 10
 libversion = '10.11.0'
 
-glib_req = '>= 2.54'
+glib_req = '>= 2.48'
 gnutls_req = '>= 2.12.0'
 gupnp_igd_req = '>= 0.2.4'
 gst_req = '>= 1.0.0'
diff --git a/nice/libnice.sym b/nice/libnice.sym
index 007be55..1e6d127 100644
--- a/nice/libnice.sym
+++ b/nice/libnice.sym
@@ -36,7 +36,6 @@ nice_agent_get_local_credentials
 nice_agent_get_remote_candidates
 nice_agent_get_selected_pair
 nice_agent_get_selected_socket
-nice_agent_get_sockets
 nice_agent_get_stream_name
 nice_agent_get_type
 nice_agent_new
